errors.go

```diff
@@ -42,10 +42,25 @@ type ErrorObject struct {
 	// Code is an application-specific error code, expressed as a string value.
 	Code string `json:"code,omitempty"`
 
+	// Source is an object containing references to the primary source of the error.
+	Source *ErrorSource `json:"source,omitempty"`
+
 	// Meta is an object containing non-standard meta-information about the error.
 	Meta *map[string]interface{} `json:"meta,omitempty"`
 }
 
+// ErrorSource is an object containing references to the primary source of the error.
```

Comment:
Thanks for the addition! Just a small suggestion: The spec suggests that only one of these attributes should be populated. I wonder if that could be part of the godoc?

```suggestion
// ErrorSource is an object containing references to the primary source of the error.
// Only one field should be populated depending on the source of the error.
```